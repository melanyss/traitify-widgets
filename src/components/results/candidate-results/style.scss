@import "~style/helpers";

.container {
  .heading {
    @include rems(16);
    @include mq(m) {
      @include rems(17);
    }
    @include mq(xl) {
      @include rems(20);
    }
    color: #767676;
    text-transform: uppercase;
    font-weight: normal;
  }
  .topPadding {
    @extend .heading;
    padding-top: 15px;
  }
  .lessBottomMargin {
    @extend .topPadding;
    margin-bottom: -10px;
  }
}
