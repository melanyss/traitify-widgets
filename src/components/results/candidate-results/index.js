import {
  faThLarge,
  faUser
} from "@fortawesome/free-solid-svg-icons";
import PropTypes from "prop-types";
import {Component} from "react";
import Icon from "lib/helpers/icon";
import withTraitify from "lib/with-traitify";
import Guide from "../../guide";
import PersonalityArchetype from "./personality-archetype";
import PersonalityDetails from "./personality-details";
import PersonalityDimensionDetails from "./personality-dimension-details";
import PersonalityDimensionColumns from "./personality-dimension-columns";
import PersonalityTraits from "../dimension-based-results/personality-traits";
import style from "./style";

class CandidateResults extends Component {
  static propTypes = {
    getOption: PropTypes.func.isRequired,
    translate: PropTypes.func.isRequired
  }
  render() {
    const {getOption, translate} = this.props;
    const allowHeaders = getOption("allowHeaders");

    return (
      <section className={style.container}>
        {allowHeaders && (
          <p className={style.heading}><Icon icon={faUser} /> {translate("personality_type")}</p>
        )}
        <PersonalityArchetype {...this.props} />
        {allowHeaders && (
          <p className={style.heading}><Icon icon={faThLarge} /> {translate("personality_details")}</p>
        )}
        <PersonalityDimensionColumns {...this.props} />
        {getOption("perspective") === "thirdPerson" && (
          <div className={style.paddingBottom}>
            {allowHeaders && (
              <p className={style.heading}><Icon icon={faThLarge} /> {translate("interview_guide_heading")}</p>
            )}
            <Guide {...this.props} />
          </div>
        )}
        {allowHeaders && (
          <p className={style.heading}><Icon icon={faThLarge} /> {translate("personality_breakdown")}</p>
        )}
        <PersonalityDimensionDetails {...this.props} />
        <p className={style.lessMarginBottom}><Icon icon={faThLarge} /> {translate("personality_advice")}</p>
        <PersonalityDetails {...this.props} />
        <p className={style.paddingTop}><Icon icon={faThLarge} /> {translate("personality_traits")}</p>
        <PersonalityTraits {...this.props} />
      </section>
    );
  }
}

export default withTraitify(CandidateResults);
