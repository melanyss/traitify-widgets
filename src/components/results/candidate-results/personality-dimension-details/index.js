import PropTypes from "prop-types";
import {Component} from "react";
import {detailWithPerspective} from "lib/helpers";
import {rgba} from "lib/helpers/color";
import TraitifyPropTypes from "lib/helpers/prop-types";
import withTraitify from "lib/with-traitify";
import DetailsList from "./details-list";
import style from "./style";

class PersonalityDimensionDetails extends Component {
  static propTypes = {
    assessment: PropTypes.shape({
      personality_types: PropTypes.arrayOf(
        PropTypes.shape({
          personality_type: PropTypes.object
        })
      )
    }),
    getOption: PropTypes.func.isRequired,
    translate: PropTypes.func.isRequired,
    ui: TraitifyPropTypes.ui.isRequired
  }
  static defaultProps = {
    assessment: null
  }
  componentDidMount() {
    this.props.ui.trigger("PersonalityDimensionDetails.initialized", this);
  }
  componentDidUpdate() {
    this.props.ui.trigger("PersonalityDimensionDetails.updated", this);
  }
  render() {
    if(!this.props.assessment) { return null; }
    const {translate} = this.props;

    const types = this.props.assessment.personality_types.sort((x, y) => {
      const xDetail = x.personality_type.details.find(({title}) => title === "Position") || {};
      const yDetail = y.personality_type.details.find(({title}) => title === "Position") || {};

      return (xDetail.body || 1) - (yDetail.body || 1);
    });

    const typesJsx = types.map((type) => {
      const {badge, details, id, level, name} = type.personality_type;
      const color = `#${badge.color_1}`;
      const perspective = this.props.getOption("perspective") || "firstPerson";
      const benefitsHeader = perspective === "firstPerson" ? translate("candidate_heading_for_benefits", {level, name}) : translate("potential_benefits");
      const options = {base: {details}, perspective};
      const benefits = details.filter(({title}) => (title === "Benefits")).map(({body}) => body);
      const pitfalls = details.filter(({title}) => (title === "Pitfalls")).map(({body}) => body);

      return (
        <li className={style.container} key={id} style={{background: rgba(color, 10), borderTop: `5px solid ${color}`, listStyle: "none"}}>
          <div className={style.side}>
            <p className={style.icon}>
              <img src={badge.image_medium} alt={`${name} ${translate("badge")}`} />
            </p>
          </div>
          <div className={style.content}>
            <h2>{name} <span style={{color}}>|</span> {level}</h2>
            {perspective === "firstPerson" && (
              <h3>{translate("candidate_heading_for_dimension", {level, name})}</h3>
            )}
            <p className={style.description}>{detailWithPerspective({...options, name: "short_description"})}</p>
          </div>
          <DetailsList detailsList={benefits} color={color} header={benefitsHeader} />
          {perspective === "thirdPerson" && (
            <DetailsList detailsList={pitfalls} color={color} header={translate("room_for_growth_and_change")} />
          )}
        </li>
      );
    });

    return (
      <div>
        {typesJsx}
      </div>
    );
  }
}

export {PersonalityDimensionDetails as Component};
export default withTraitify(PersonalityDimensionDetails);
