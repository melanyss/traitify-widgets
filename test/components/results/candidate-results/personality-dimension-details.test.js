import {Component} from "components/results/candidate-results/personality-dimension-details";
import ComponentHandler from "support/component-handler";
import assessment from "support/json/assessment/dimension-based.json";

jest.mock("lib/with-traitify", () => ((value) => value));

describe("PersonalityDimensionDetails", () => {
  let props;

  beforeEach(() => {
    props = {
      assessment,
      translate: jest.fn().mockName("translate"),
      getOption: jest.fn().mockName("getOption"),
      type: assessment.personality_types[0],
      ui: {
        current: {},
        off: jest.fn().mockName("off"),
        on: jest.fn().mockName("on"),
        trigger: jest.fn().mockName("trigger")
      }
    };
  });

  describe("callbacks", () => {
    it("triggers initialization", () => {
      const component = new ComponentHandler(<Component {...props} />);

      expect(props.ui.trigger).toHaveBeenCalledWith("PersonalityDimensionDetails.initialized", component.instance);
    });

    it("triggers update", () => {
      const component = new ComponentHandler(<Component {...props} />);
      component.updateProps();

      expect(props.ui.trigger).toHaveBeenCalledWith("PersonalityDimensionDetails.updated", component.instance);
    });
  });

  it("renders component", () => {
    const component = new ComponentHandler(<Component {...props} />);
    expect(component.tree).toMatchSnapshot();
  });

  it("renders component with headers", () => {
    props.translate.mockReturnValue("Detail Header");
    props.options = {allowHeaders: true};

    const component = new ComponentHandler(<Component {...props} />);
    expect(component.tree).toMatchSnapshot();
  });

  it("renders component with pitfalls and headers", () => {
    props.translate.mockReturnValue("Detail Header");
    props.getOption.mockReturnValue("thirdPerson");

    const component = new ComponentHandler(<Component {...props} />);
    expect(component.tree).toMatchSnapshot();
  });
});
