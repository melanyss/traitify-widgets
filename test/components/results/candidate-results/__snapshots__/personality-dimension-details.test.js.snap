// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalityDimensionDetails renders component 1`] = `null`;

exports[`PersonalityDimensionDetails renders component with pitfalls 1`] = `null`;
